src/osd/OSD.cc

```diff
@@ -4008,6 +4008,25 @@ void OSD::load_pgs()
     // read pg state, log
     pg->read_state(store);
 
+    if (pg->dne())  {
+      dout(10) << "load_pgs " << *it << " deleting dne" << dendl;
+      pg->ch = nullptr;
+      service.pg_remove_epoch(pg->pg_id);
+      pg->unlock();
+      {
+	// Delete pg
+	RWLock::WLocker l(pg_map_lock);
+	auto p = pg_map.find(pg->get_pgid());
```

Comment:
if a pg is returned by `_open_lock_pg()`, i think we can assume that this pg is always added to `pg_map`, am i right? so this check is not necessary, and can be replaced with an assert() i guess?

Reply:
True

Reply:
@tchaikov Changed to an assert